core/instantiate.ml

```diff
@@ -2,6 +2,11 @@ open CommonTypes
 open Utility
 open Types
 
+exception ArityMismatch of (int * int)
+
+let internal_error message =
```

Comment:
Excellent idea.

Reply:
I ended up refactoring this so that the exception wasn't constructed directly in each file, but one recurring thing I noticed was that I had to specify the same filename multiple times in a single file, which felt wrong. 

So each file needing `internal_error` still has it defined, which calls `internal_error` in `errors.ml`.

Reply:
Yes, that sounds good.